sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -543,16 +596,38 @@ def _load_events(self, events):
                 )
 
     @property
-    def size_in_bytes(self):
-        # type: () -> int
+    def message(self) -> Union["BatchMessage", LegacyBatchMessage]:
+        """
+        DEPRECATED: Get the underlying uamqp.BatchMessage or LegacyBatchMessage.
+         This is deprecated and will be removed in a later release.
+        :rtype: uamqp.BatchMessage or LegacyBatchMessage
+        """
+        warnings.warn("The `message` property is deprecated and will be removed in future versions.", DeprecationWarning)
+        if not self._uamqp_message:
+            message = AmqpAnnotatedMessage(message=Message(*self._message))
+            self._uamqp_message = LegacyBatchMessage(
+                message, to_outgoing_amqp_message=PyamqpTransport().to_outgoing_amqp_message
```

Comment:
This method only takes a pyamqp Message, I believe? so, 2 cases:
1) The transport is UamqpTransport. In which case, self._uamqp_message will already be set in the constructor and this won't be built here. (And I think passing in the uamqp Message to this would result in an error anyway?)
2) The transport is PyamqpTransport. In which case, self._uamqp_message will be set to None in the constructor, and this will always need the PyamqpTransport.to_outgoing_amqp_message).

Does this sound reasonable?

Reply:
Yes I think it seems reasonable... though does this property work okay if I'm using the uamqp transport? As it seems to be creating an instance of the pyamqp Message on line 607

Reply:
TL;DR: yes, this property works for both uamqp transport and pyamqp transport. `test_legacy_message` in `unittest/test_event_data.py` checks for this. When it's using the UamqpTransport, the pyamqp Message will never be created - `self._uamqp_message` will be set in the constructor and that if block will not be entered.

More info:
upon further digging, the following properties never worked for uamqp.BatchMessage: `get_data(), settled, delivery_annotations, encode_message(), footer, get_message()`. (Tested with EH 5.9.0 and uamqp 1.5.2.) `print(event_data_batch.message)` also resulted in an AttributeError.

It looks BatchMessage was inheriting a few of these attributes from Message, but these were never actually tested. (For ex. [`get_data()` checks for `self._message`](https://github.com/Azure/azure-uamqp-python/blob/4b1cb844e1c62366600e3770c3dde4a7334cb5d9/uamqp/message.py#L491), but BatchMessage doesn't have a `_message` attribute.)

Given that these never worked, I'm assuming that customers never used/called these properties. So, I only added tests to check that the working properties still work. (Although, the properties that don't work in the uamqp BatchMessage actually do work in the LegacyBatchMessage.)